### DISCUSSION TITLE
Can pocketbase/dbx supports AfterFind, BeforeCreate hooks (eager loading) ?

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#1436] [Q&A] [answered]
Hi ~
Can I add more value after I query something or before create ?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand the question.
The available hooks could be found in https://pocketbase.io/docs/event-hooks/.

Could you provide a code sample what are you trying to do?

**Reply by hungcrush** [reply]
I'm sorry for my english. please take a look my capture. After I query, FullUrl will be added.

<img width="513" alt="CleanShot 2022-12-30 at 12 30 46@2x" src="https://user-images.githubusercontent.com/8098916/210037899-66f79cb2-faf3-413d-929d-4c39dee8a961.png">

This is really important when I want to customize data after query db

**Reply by ganigeorgiev** [reply]
I haven't tested it, but you should be able to make use of the `PostScanner` interface, something like:

```go
func (m *MediaFile) PostScan() error {
    m.BaseModel.PostScan()

    // do something after db scan...

    return nil
}
```

**Reply by hungcrush** [reply]
Wowwww... it works. That's what I want. 

Thank you very very much.. I love you.
